src/puppetlabs/pcp/client.clj

```diff
@@ -325,6 +325,14 @@
     (.start client)
     client))
 
+(defn ^:private append-client-type
+  "Append client type to a ws connection Uri, accounting for possible trailing
+   slash."
+  [url type]
+  (if (= (last url) \/)
```

Comment:
I'm not sure if there is any optimization elsewhere, but this implementation does iterate: https://github.com/clojure/clojure/blob/clojure-1.8.0/src/clj/clojure/core.clj#L262-L270